staging/src/k8s.io/api/node/v1beta1/types.go

```diff
@@ -48,6 +49,25 @@ type RuntimeClass struct {
 	// The Handler must conform to the DNS Label (RFC 1123) requirements, and is
 	// immutable.
 	Handler string `json:"handler" protobuf:"bytes,2,opt,name=handler"`
+
+	// Topology specifies the scheduling constrains that are necessary to assign
+	// pods to the right node
+	// +optional
+	Topology Topology `json:"topology,omitempty" protobuf:"bytes,3,opt,name=topology"`
+}
+
+// Topology specifies the structure of scheduling constrains for the runtime class
+type Topology struct {
+	// nodeSelector selects the set of nodes that support this RuntimeClass.
+	// Pods using this RuntimeClass can only be scheduled to a node matched by
+	// this selector. The nodeSelector is intersected (AND) with a pod's other
+	// node affinity or node selector requirements.
+	// +optional
+	NodeSelector v1.NodeSelector `json:"nodeSelector,omitempty" protobuf:"bytes,1,rep,name=nodeSelector"`
```

Comment:
Needs to be a pointer (all optional fields should be a pointer, unless they're a slice/map)